src/app/login/pattern.component.html

```diff
@@ -0,0 +1,31 @@
+<div class="container-fluid text-center">
+        <form role="form">
+            <div class="row header">
+                mPIN
+            </div>
+
+            <div class="row pin-input">
+                <div class="form-group ">
+                    <label for="pin">Your unique PIN</label>
+                    <input class="form-control input-lg" id="pin" [value]='pin' type="password" maxlength="4" placeholder="Enter 4 digits pin" readonly>
+                </div>
+            </div>
+
+            <div class="pin-keys">
+                <div *ngFor='#num of [1,4,7]' class="row">
+                    <button type="button" (click)='append(num)' class="btn btn-circle btn-default">{{num}}</button>

```

Comment:
Events should be binded in Angular JS way. Also use, data-ng-bind instead of using {{num}}